rfcs/1440_generic_planner_node.md

```diff
@@ -0,0 +1,482 @@
+
+The planner node will be responsible for creating the required pseudo nodes as
+well. In the specific cases of HTTP- and gRPC pseudo nodes each pseudo node will
+send all invocations over a single dedicated invocation channel. The planner
+node will, by default, have the handle to receive these invocations and
+therefore be responsible for routing them to the right Wasm nodes based on the
+associated labels (specifically the label associated with the request receiver
+channel should match the label of the target Wasm node). Since it would be
+possible for multiple Wasm nodes to share the same label, the planner node would
+need an indication of which Wasm node should receive invocations with a specific
+label on the request receiver channel.
+
+If more advanced routing is required, the planner node could forward the
+invocation receiver channel to another node that can do the routing. The problem
+in this case is that the new node would have to be responsible for creating any
+per-label nodes as well if these nodes must be dynamically created, seeing that
+the planner node will no longer be able to see incoming invocations.
+
+Questions:
+
+- Would there be a requirement for multiple gRPC server pseudo nodes running on
+  different ports to each have a different configuration for Invocation routing?
+- Should we use special handling of channel types to determine the targets for
+  routing, or should we mark specific nodes as targeted invocation receivers?
+- Should we support overriding all routing so that the planner node still
+  receives all invocations, but always forward it to a single specified node
+  that is responsible for the routing?
+
+### Manifest File
+
+The manifest file will be passed to the application as part of the initial
+configuration map as a TOML file. The assumption is that each application will
+have one unique node graph manifest file, which will be maintained by the
+developer of the application. The node graph manifest should be consistent with
+the application configuration file and the defense-in-depth restrictions. As
+part of the future work we plan on building analysis tools to validate that this
+is the case.
+
+The file will be deserialised into the following structure:
+
+```rust
+
+/// A static node definition.
+struct Node {
+  /// Unique name for the node.
+  name: String,
+
+  /// The label associated with the node.
+  ///
+  /// Note The `Label` type is the same type defined in label.proto.
+  label: Label,
+
+  /// The keys of the files from the config map that must be passed to the node
+  /// as part of the initialisation message. This will be used to generate a
+  /// filtered view of the map that will be sent as part of the initialisation
+  /// message.
+  config_files: Vec<String>,
+
+  /// Whether this node should be the target node for all gRPC invocations
+  /// targeted at the specified label. This is only used if the planner node is
+  /// also responsible for invocation routing.
```

Comment:
This seems too specific about roles, also not sure how it would allow arbitrary logic without having to reinvent Wasm at the manifest level.

Reply:
The reason for this RFC is to not allow arbitrary logic during node creation to enable easier static analysis.

Reply:
Could we have at least a couple of more fleshed out examples? e.g. the PIR, the chat app, and the aggregator? To make sure that it is indeed possible to represent them in this way.